From mark.a.allyn@intel.com  Wed Feb 17 14:48:36 2010
From: Mark Allyn <mark.a.allyn@intel.com>
Date: Fri,  5 Feb 2010 10:53:34 -0800
Subject: Staging: rar_register: add suspend and resume functions
To: linux-kernel@vger.kernel.org, alan@linux.intel.com, charles.f.johnson@intel.com, greg@kroah.com
Cc: Mark Allyn <mark.a.allyn@intel.com>
Message-ID: <1265396014-16107-1-git-send-email-mark.a.allyn@intel.com>


Add suspend and resume functions (which are currently stubs
returning -ENOSYS)

Signed-off-by: Mark Allyn <mark.a.allyn@intel.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/staging/rar_register/rar_register.c |   15 ++++++++++++++-
 1 file changed, 14 insertions(+), 1 deletion(-)

--- a/drivers/staging/rar_register/rar_register.c
+++ b/drivers/staging/rar_register/rar_register.c
@@ -506,6 +506,17 @@ int register_rar(int (*callback)(void *y
 }
 EXPORT_SYMBOL(register_rar);
 
+/* Suspend - returns -ENOSYS */
+static int rar_suspend(struct pci_dev *dev, pm_message_t state)
+{
+	return -ENOSYS;
+}
+
+static int rar_resume(struct pci_dev *dev)
+{
+	return -ENOSYS;
+}
+
 /*
  * This function registers the driver with the device subsystem (
  * either PCI, USB, etc).
@@ -582,7 +593,9 @@ const struct pci_device_id *my_id_table
 static struct pci_driver rar_pci_driver = {
 	.name = "rar_register_driver",
 	.id_table = rar_pci_id_tbl,
-	.probe = rar_probe
+	.probe = rar_probe,
+	.suspend = rar_suspend,
+	.resume = rar_resume
 };
 
 static int __init rar_init_handler(void)
